types/validate_data.go

```diff
@@ -3,20 +3,31 @@ package types
 import (
 	"strconv"
 
-	panaceatypes "github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	datadealtypes "github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	datapooltypes "github.com/medibloc/panacea-core/v2/x/datapool/types"
 )
 
-func NewUnsignedDataValidationCertificate(dealIdStr string, dataHash []byte, encryptedDataUrl []byte, requesterAddress, dataValidatorAddress string) (panaceatypes.UnsignedDataValidationCertificate, error) {
+func NewUnsignedDataValidationCertificateOfDataDeal(dealIdStr string, dataHash []byte, encryptedDataUrl []byte, requesterAddress, dataValidatorAddress string) (datadealtypes.UnsignedDataValidationCertificate, error) {
```

Comment:
Instead of having `*DataDeal` suffix, I would recommend to have two separated packaged:
```
types
├── datadeal
│   └── validate_data.go
├── datapool
│   └── validate_data.go
```
That's what our service packages already do:
```
server/service
├── datadeal
│   ├── service.go
│   └── validate_data_handler.go
├── datapool
│   ├── download_data_handler.go
│   ├── service.go
│   └── validate_data_handler.go
```

Reply:
To be honest, I don't think that the `validate_data.go` file is necessary. This file contains only simple struct constructors without complex preprocessor/postprocessor. Also, those constructors are used only in one place. Then, I would just create a struct directly when I need:
```
c := datadealtypes.UnsignedDataValidationCertificate{
...
}
```

Reply:
Great.
I just fixed it.